include/mpp/algorithm/forward_substitution.hpp

```diff
@@ -99,10 +117,27 @@ namespace mpp
 				}
 			}
 
-			return matrix<To, RowsExtent, ColumnsExtent>{ a.rows(),
-				1,
-				forward_subst_on_buffer<To, RowsExtent, ColumnsExtent, Check>(a.data(), b.data(), a.rows()),
-				unsafe };
+			const auto rows = a.rows();
+
+			using x_mat_t = matrix<To, RowsExtent, ColumnsExtent, ToAllocator>;
+			auto&& x_buf  = forward_subst_on_buffer<Check,
+                RowsExtent,
+                ColumnsExtent,
+                typename std::allocator_traits<ToAllocator>::template rebind_alloc<default_floating_type>>(a.data(),
+                b.data(),
+                rows,
+                alloc_args...);
```

Comment:
When `To` is != `default_floating_type`, it'll fail to compile when passing an allocator of type `To` because the value type conflicts with the internal buffer's value type.